pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunction.java

```diff
@@ -107,6 +108,12 @@ void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResultHolder
    */
   ColumnDataType getIntermediateResultColumnType();
 
+  /**
+   * Returns the {@link ColumnDataType} of the final result.
+   * <p>This column data type is used in the {@link ResultTable::_dataSchema}
```

Comment:
(nit) remove the link to ResultTable. The javadoc should not access private fields.
Can be replaced with `This column data type is used for constructing the result table`